connect/runtime/src/main/java/org/apache/kafka/connect/storage/OffsetStorageWriter.java

```diff
@@ -120,6 +120,13 @@ public synchronized boolean beginFlush() {
         return true;
     }
 
+    /**
+     * @return whether there's anything to flush right now.
+     */
+    public boolean willFlush() {
```

Comment:
synchronized?  All access to `data` seems to be in synchronizedd

Reply:
Good catch, thanks!